### ISSUE TITLE
Duplicate ErrorCode.WRN_ConvertingNullableToNonNullable warning is reported at the same location

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#30938]
```
        [Fact]
        public void Test()
        {
            var source = @"
class Outer
{
    void M1(object x1, object? y1)
    {
        x1 = (object)y1;
    }
}
";

            CreateCompilation(source, options: WithNonNullTypesTrue()).VerifyDiagnostics(
                // (6,14): warning CS8600: Converting null literal or possible null value to non-nullable type.
                //         x1 = (object)y1;
                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(object)y1").WithLocation(6, 14),
                // (6,14): warning CS8600: Converting null literal or possible null value to non-nullable type.
                //         x1 = (object)y1;
                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(object)y1").WithLocation(6, 14)
                );
        }
```

One is reported when we analyze conversion, another is reported when we analyze the assignment. Conversion preserves nullable state.

**Comment by cston** [comment]
See also https://github.com/dotnet/roslyn/issues/29960.
